src/main/java/com/aws/greengrass/deployment/ConnectivityValidator.java

```diff
@@ -0,0 +1,482 @@
+    }
+
+    private void checkHttpConnection(long timeout) throws InterruptedException,
+            ComponentConfigurationValidationException {
+        ListThingGroupsForCoreDeviceRequest request = ListThingGroupsForCoreDeviceRequest.builder()
+                .coreDeviceThingName(Coerce.toString(deviceConfiguration.getThingName())).build();
+        long tryUntil = System.currentTimeMillis() + timeout;
+        while (System.currentTimeMillis() < tryUntil) {
+            try {
+                greengrassV2DataClient.listThingGroupsForCoreDevice(request);
+                return;
+            } catch (GreengrassV2DataException | SdkClientException e) {
+                Thread.sleep(TimeUnit.SECONDS.toMillis(5));
+            }
+        }
+        throw new ComponentConfigurationValidationException(
+                "HTTP client failed to connect to IoT Core using new configurations",
+                DeploymentErrorCode.FAILED_TO_RECONNECT);
+    }
+
+    private void checkMqttConnection(long timeout) throws InterruptedException,
+            ComponentConfigurationValidationException {
+        long tryUntil = System.currentTimeMillis() + timeout;
+        while (System.currentTimeMillis() < tryUntil) {
+            try {
+                connection.connect().get();
+                return;
+            } catch (MqttException | ExecutionException e) {
+                Thread.sleep(TimeUnit.SECONDS.toMillis(2));
+            }
+        }
+        throw new ComponentConfigurationValidationException(
+                "MQTT client failed to connect to IoT Core using new configurations",
+                DeploymentErrorCode.FAILED_TO_RECONNECT);
+    }
+
+    private void createMqttConnectionBuilder(Map<String, Object> nucleusConfig)
+            throws MqttConnectionProviderException {
+        builder = securityService.getDeviceIdentityMqttConnectionBuilder();
+
+        // get mqtt values from nucleus config to construct a MQTT client
+        int pingTimeoutMs = tryGetMqttPingTimeoutFromNewConfig(nucleusConfig);
+        int keepAliveMs = tryGetMqttKeepAliveMsFromNewConfig(nucleusConfig);
+        if (keepAliveMs != 0 && keepAliveMs <= pingTimeoutMs) {
+            throw new MqttException(String.format("%s must be greater than %s",
+                    MQTT_KEEP_ALIVE_TIMEOUT_KEY, MQTT_PING_TIMEOUT_KEY));
+        }
+        String rootCaPath = Coerce.toString(deviceConfiguration.getRootCAFilePath());
+        String clientId = Coerce.toString(deviceConfiguration.getThingName());
+        String endpoint = tryGetIoTDataEndpointFromNewConfig(nucleusConfig);
+        int port = tryGetMqttPortFromNewConfig(nucleusConfig);
+        int operationTimeout = tryGetMqttOperationTimeoutFromNewConfig(nucleusConfig);
+        int socketTimeout = tryGetMqttSocketTimeoutFromNewConfig(nucleusConfig);
+
+        // start building mqtt connection
+        builder.withCertificateAuthorityFromPath(null, rootCaPath)
+                .withClientId(clientId)
```

Comment:
this is guaranteed to fail. You're using a client ID which will already be in use.

Reply:
will append a suffix